File: CodeForces/834/C.cpp
#include <bits/stdc++.h>

#define lli long long int
#define pb push_back
#define pii pair <int, int>
#define pll pair <lli, lli>
#define _F first
#define _S second
#define mset(x) memset(x, 0, sizeof(x))
#define fastio() ios_base::sync_with_stdio(0); cin.tie(NULL)

using namespace std;

int check(lli, lli, lli, lli);

int main()
{
	fastio();

	int n;
	cin >> n;

	for(int i = 0; i < n; i++)
	{
		lli x, y;
		cin >> x >> y;
		lli p = x * y;
		lli a = cbrt(p);

		if(	check(p, a - 1, x, y) or
			check(p, a + 0, x, y) or
			check(p, a + 1, x, y))
			cout << "Yes" << "\n";
		else
			cout << "No" << "\n";
	}
	
	return (0-0);	
}

int inline check(lli p, lli c, lli x, lli y)
{
	if(p == (c * c * c))
	{
		if(x % c or y % c)
			return 0;
		else
			return 1;
	}
	else
		return 0;
}

File: CodeForces/834/C.py
def cuberoot(x):
	return int(x ** (1. / 3))

def check(x, y, c):
	p = x * y

	if not c:
		return 0

	if p != c ** 3:
		return 0

	if x % c or y % c:
		return 0

	return 1

n = int(input())

for _ in range(n):
	x, y = map(int, input().split())

	c = cuberoot(x * y)

	if check(x, y, c - 1) or check(x, y, c) or check(x, y, c + 1):
		print("Yes")
	else:
		print("No")